native/cocos/2d/renderer/RenderDrawInfo.h

```diff
@@ -0,0 +1,179 @@
+    void setIsMeshBuffer(bool isMeshBuffer);
+    inline Material* getMaterial() const { return _material; }
+    void setMaterial(Material* material);
+    inline gfx::Texture* getTexture() const { return _texture; }
+    void setTexture(gfx::Texture* texture);
+    inline uint32_t getTextureHash() const { return _textureHash; }
+    void setTextureHash(uint32_t textureHash);
+    inline gfx::Sampler* getSampler() const { return _sampler; }
+    void setSampler(gfx::Sampler* sampler);
+    inline uint32_t getBlendHash() const { return _blendHash; }
+    void setBlendHash(uint32_t blendHash);
+    inline scene::Model* getModel() const { return _model; }
+    void setModel(scene::Model* model);
+
+    void setRender2dBufferToNative(uint8_t* buffer, uint8_t stride, uint32_t size, uint8_t type);
+
+    inline Batcher2d* getBatcher() const { return _batcher; }
+    void setBatcher(Batcher2d* batcher);
+    const ArrayBuffer& getAttrSharedBufferForJS() const;
+
+    inline Render2dLayout* getRender2dLayout(uint32_t dataOffset) {
+        return reinterpret_cast<Render2dLayout*>(_sharedBuffer + dataOffset * sizeof(float_t));
+    }
+
+    inline uint8_t getStride() { return _stride; }
+    inline uint32_t getSize() { return _size; }
+    inline uint8_t getDrawType() { return _drawType; }
+
+    gfx::InputAssembler* requestIA(gfx::Device* device);
+    void uploadBuffers();
+    void resetMeshIA();
+
+
+private:
+    gfx::InputAssembler* _initIAInfo(gfx::Device* device);
+
+    Batcher2d* _batcher{nullptr};
+
+    uint8_t* _sharedBuffer{nullptr};
+    uint8_t _stride{0};
+    uint32_t _size{0};
+
+    DrawInfoAttrLayout _drawInfoAttrLayout{};
+    se::Object* _seArrayBufferObject{nullptr};
+    ArrayBuffer::Ptr _attrSharedBuffer{nullptr};
+
+    index_t _bufferId{0};
+    index_t _accId{0};
+    uint32_t _vertexOffset{0};
+
+    uint32_t _indexOffset{0};
+
+    float_t* _vbBuffer{nullptr};
+    uint16_t* _ibBuffer{nullptr};
+    float_t* _vDataBuffer{nullptr};
+    uint16_t* _iDataBuffer{nullptr};
+    UIMeshBuffer* _meshBuffer{nullptr};
+
+    uint32_t _vbCount{0};
+    uint32_t _ibCount{0};
+
+    Node* _node{nullptr};
+
+    bool _vertDirty{false};
+    uint8_t _drawType{0};
+
+    scene::Model* _model{nullptr};
+
+    ccstd::hash_t _dataHash{0};
+    uint32_t _stencilStage{0};
+    bool _isMeshBuffer{false};
+    Material* _material{nullptr};
+    gfx::Texture* _texture{nullptr};
+    uint32_t _textureHash{0};
+    gfx::Sampler* _sampler{nullptr};
+
+    uint32_t _blendHash{0};
+
+    gfx::InputAssemblerInfo _iaInfo;
+    ccstd::vector<gfx::Attribute> _attributes{
+        gfx::Attribute{gfx::ATTR_NAME_POSITION, gfx::Format::RGB32F},
+        gfx::Attribute{gfx::ATTR_NAME_TEX_COORD, gfx::Format::RG32F},
+        gfx::Attribute{gfx::ATTR_NAME_COLOR, gfx::Format::RGBA32F},
+    };
+
+    ccstd::vector<gfx::InputAssembler*> _iaPool{};
+    uint32_t _nextFreeIAHandle{0};
+    gfx::Buffer* vbGFXBuffer{nullptr};
+    gfx::Buffer* ibGFXBuffer{nullptr};
```

Comment:
- please add `_` for variables
- where to delete them

Reply:
@LinYunMo

Reply:
delete them on destroy